src/main/java/org/javarosa/core/model/FormDef.java

```diff
@@ -1428,15 +1426,21 @@ public void initialize(boolean newInstance, InstanceInitializationFactory factor
      *                    false if it is using an existing IDataModel
      * @param locale      The default locale in the current environment, if provided. Can be null
      *                    to rely on the form's internal default.
+     * @param isReadOnly  If we are in read only mode and only wants to view form
+     * @param remoteInstanceFetcher utility to initialize remote instances over internet, can be null
      */
-    public void initialize(boolean newInstance, boolean isCompletedInstance,
-                           InstanceInitializationFactory factory, String locale, boolean isReadOnly) {
+    public void initialize(boolean newInstance, boolean isCompletedInstance, InstanceInitializationFactory factory,
+                           String locale, boolean isReadOnly, @Nullable RemoteInstanceFetcher remoteInstanceFetcher) throws RemoteInstanceFetcher.RemoteInstanceException {
         for (Enumeration en = formInstances.keys(); en.hasMoreElements(); ) {
             String instanceId = (String)en.nextElement();
             DataInstance instance = formInstances.get(instanceId);
             formInstances.put(instanceId, instance.initialize(factory, instanceId));
         }
 
+        if (remoteInstanceFetcher != null) {
+            tryAttachingRemoteInstances(remoteInstanceFetcher);
```

Comment:
Just for my understanding, why is this happening in the formDef? Surely this should happen prior to this step?